src/orion/core/io/space_builder.py

```diff
@@ -289,60 +322,114 @@ def _build_from_config(self, config_path):
                         raise ValueError(error_msg) from exc
 
     def _build_from_args(self, cmd_args):
+        """Build templates from arguments found in the original cli.
+
+        Rules for namespacing:
+           1. Prefix ``'/'`` is given to parameter dimensions.
+           2. Prefix ``'$'`` is given to substitutions from exposed properties.
+           3. Prefix ``'_'`` is given to arguments which do not interact with.
+              Suffix is a unique integer.
+           4. ``'config'`` is given to user's script configuration file template
+
+        User script configuration file argument is treated specially, trying to
+        recognise a ``--config`` option or by checking the first positional
+        argument, if not found elsewhere. Only one is allowed.
+
+        .. note:: Currently positional arguments cannot define a parameter
+           dimension. And I don't think they will; it wouldn't be good for
```

Comment:
We should not use first person singular in docstrings. We can't use positional arguments because they don't have a name assigned to. I think that's enough justification.